actions/action.go

```diff
@@ -62,8 +65,27 @@ func NewOutputs(uri string, latestVersion *semver.Version, additionalOutputs Out
 	return outputs, nil
 }
 
-func (o Outputs) Write(writer io.Writer) {
+func (o Outputs) Write() {
+	outputFileName, ok := os.LookupEnv("GITHUB_OUTPUT")
+	if !ok {
+		panic(errors.New("GITHUB_OUTPUT is not set, see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter"))
+	}
+	writer, err := os.OpenFile(outputFileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
[`os.Create`](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/os/file.go;l=327) passes the `O_TRUNC` flag, we pass `O_APPEND` instead.

Basically the same as `>> "$GITHUB_OUTPUT"` vs `> "$GITHUB_OUTPUT"` in bash would be. In their examples [GitHub uses append](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#examples).
I assume, if we truncate the file we would delete outputs that were created elsewhere. This might lead to subtle bugs, as GitHub Actions don't fail on missing values in their interpolation syntax.
So IMHO this is the better option, even if it is less readable.

My suggestions:
- Either I add a comment explaining the difference to `os.Open`
- or I move the `os.OpenFile` call to a new helper function called `Append`, with a doc comment that it works like `os.Open`, but uses the append instead of the truncate flag.

What would you prefer?

Reply:
Ah ha. Ok. That makes perfect sense, don't want to truncate the file and wipe out previously written output.

To clarify the code I don't have a strong preference, but given a choice, I'd say let's go with the `Append` method.

Reply:
I called the method `createOrAppend`:
1. I guess it's better private, therefore lowercase
1. `append` would collide with the built-in array append function.